Assets/HoloToolkit/Boundary/Scripts/BoundaryManager.cs

```diff
@@ -0,0 +1,162 @@
+            SetBoundaryRendering();
+
+            // Render the floor based on if you are in editor or immersive device.
+            RenderFloorQuad();
+        }
+
+        private void RenderFloorQuad()
+        {
+            if (FloorQuad != null && HolographicSettings.IsDisplayOpaque)
+            {
+                floorQuadInstance = Instantiate(FloorQuad);
+                floorQuadInstance.transform.SetParent(gameObject.transform.parent);
+
+#if UNITY_EDITOR
+                // So the floor quad does not occlude in editor testing, draw it lower.
+                floorQuadInstance.transform.localPosition = new Vector3(0, -3, 0);
+#else
+                // Inside immersive headset draw floor quad at Y value of dimensions.
+                Vector3 dimensions;
+                // TODO: BUG: Unity: configured bool always returns false.
+                // TODO: BUG: Unity: TryGetDimensions does not return true either.
+                //if (UnityEngine.Experimental.XR.Boundary.configured &&
+                //    UnityEngine.Experimental.XR.Boundary.TryGetDimensions(out dimensions))
+
+                if (UnityEngine.Experimental.XR.Boundary.TryGetDimensions(out dimensions))
+                {
+                    if (dimensions != null)
+                    {
+                        Debug.Log("Drawing floor at dimensions Y.");
+                        // Draw the floor at boundry Y.                    
+                        floorQuadInstance.transform.localPosition = new Vector3(0, dimensions.y, 0);
+                    }
+                }
+                else
+                {
+                    Debug.Log("Drawing floor at 0,0,0.");
+                    // Draw the floor at 0,0,0.
+                    floorQuadInstance.transform.localPosition = Vector3.zero;
+                }
+#endif
+                floorQuadInstance.SetActive(true);
+            }
+        }
+
+        /// <summary>
+        /// Pass in the game object's position to check if it's within 
+        /// the specified boundary space.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        public bool ContainsObject(Vector3 gameObjectPosition)
+        {
+            // TODO: BUG: Unity
+            //if (!UnityEngine.Experimental.XR.Boundary.configured)
+            //{
+            //    Debug.Log("Boundary not configured.");
+            //    return;
+            //}
+
+            // Get all the bounds setup by the user.
+            List<Vector3> boundaryGeometry = new List<Vector3>();
+            if (UnityEngine.Experimental.XR.Boundary.TryGetGeometry(boundaryGeometry))
+            {
+                if (boundaryGeometry != null)
+                {
+                    // Create a UnityEngine.Bounds volume with those values.
+                    foreach (Vector3 boundaryGeo in boundaryGeometry)
+                    {
+                        boundaryBounds.Encapsulate(boundaryGeo);
```

Comment:
Do we expect the bounds to change during an app's lifetime? It seems like it could be expensive to regenerate the bounds every time this method is called. Could it be set in Start()? Or perhaps poll in Update until we get a response, if we expect it might not be immediately available.